lib/common/schemas.c

```diff
@@ -650,6 +652,51 @@ validate_xml(xmlNode *xml_blob, const char *validation, gboolean to_logs)
 }
 
 #if HAVE_LIBXSLT
+
+/* see crm_xml_error */
+static void
+crm_xslt_err(void *ctx, const char *msg, ...)
+G_GNUC_PRINTF(2, 3);
+
+static void
+crm_xslt_err(void *ctx, const char *msg, ...)
+{
+    int len = 0;
+    va_list args;
+    char *buf = NULL;
+    static int buffer_len = 0;
+    static char *buffer = NULL;
+
+    va_start(args, msg);
+    len = vasprintf(&buf, msg, args);
+
+    if (strchr(buf, '\n')) {
+        buf[len - 1] = 0;
+        if (buffer) {
+            crm_err("XSLT reports: %s%s", buffer, buf);
+            free(buffer);
+        } else {
+            crm_err("XSLT reports: %s", buf);
+        }
+        buffer = NULL;
+        buffer_len = 0;
+
+    } else if (buffer == NULL) {
+        buffer_len = len;
+        buffer = buf;
+        buf = NULL;
+
+    } else {
+        buffer = realloc(buffer, 1 + buffer_len + len);
+        memcpy(buffer + buffer_len, buf, len);
+        buffer_len += len;
+        buffer[buffer_len] = 0;
+    }
```

Comment:
To avoid any hassles with __FILE__, etc., I am leaning towards the common
base being a parametrized macro in `crm/common/xml.h`.
Ack for `CIB upgrade`,